client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx

```diff
@@ -99,14 +99,11 @@ export default function WPCheckoutOrderReview( {
 	const cartKey = useCartKey();
 	const { responseCart, removeCoupon, couponStatus } = useShoppingCart( cartKey );
 	const isPurchaseFree = responseCart.total_cost_integer === 0;
-	const isDesktop = useViewportMatch( 'mobile', '>=' );
 	const reduxDispatch = useDispatch();
 
 	useEffect( () => {
-		const experimentCheck = isDesktop
-			? 'registration_email_only_desktop_random_usernames'
-			: 'registration_email_only_mobile_random_usernames';
-		let shouldCheck = true;
+		const experimentCheck = retrieveExperimentName();
+		let shouldCheck = experimentCheck?.length > 0;
 		loadExperimentAssignment( experimentCheck ).then( ( experimentObject ) => {
```

Comment:
client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx:107:29 - error TS2345: Argument of type 'string | null' is not ass
ignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.

Reply:
`loadExperimentAssignment()` requires a string. There's various ways to resolve this; here's one:

```suggestion
		loadExperimentAssignment( experimentCheck ?? '' ).then( ( experimentObject ) => {
```

Reply:
Thank you @sirbrillig , looks like your suggestions fixed it!